packages/glsp-playwright/src/integration/vscode/vscode.integration.ts

```diff
@@ -0,0 +1,185 @@
+    userDataDir: string;
+}
+
+/**
+ * The {@link VSCodeIntegration} provides the glue code for working
+ * with the VSCode version of the GLSP-Client.
+ *
+ * The integration will:
+ *
+ * - prefix the root selector to the correct frame, as webviews are used in VSCode
+ * - start the VSCode instance (i.e., Electron) with different run configurations
+ * - check if the extension is already installed
+ * - wait for the GLSP-Client
+ *
+ * **Note**
+ *
+ * Run configurations allow starting multiple Electron instances in parallel
+ * and the configuration will be saved in the temp folder.
+ */
+export class VSCodeIntegration extends Integration {
+    readonly type: IntegrationType = 'VSCode';
+    workbenchActivitybar: VSCodeWorkbenchActivitybar;
+
+    protected runConfig: VSCodeRunConfig;
+    protected electronApp: ElectronApplication;
+    protected storage: VSCodeStorage.Storage;
+
+    constructor(protected readonly options: VSCodeIntegrationOptions) {
+        super();
+    }
+
+    override async initialize(): Promise<void> {
+        const storagePath = this.options.storagePath;
+        const storage = await VSCodeStorage.read(storagePath);
+        if (VSCodeStorage.is(storage)) {
+            this.storage = storage;
+        } else {
+            throw Error(`Provided storage in "${storagePath}" was not a valid vscode storage. ${JSON.stringify(storage, null, 2)}`);
+        }
+    }
+
+    get page(): Page {
+        return this.electronApp.windows()[0];
+    }
+
+    override prefixRootSelector(selector: string): Locator {
+        return this.page.frameLocator('iframe').frameLocator('iframe').locator(selector);
+    }
+
+    override async close(): Promise<void> {
+        await this.electronApp.close();
+        await fs.rm(this.runConfig.runFolder, { recursive: true, force: true });
+    }
+
+    protected override async beforeLaunch(): Promise<void> {
+        this.runConfig = await this.createRunConfiguration(this.options, this.storage);
+    }
+
+    protected override async launch(): Promise<void> {
+        this.electronApp = await electron.launch({
+            executablePath: this.storage.vscodeExecutablePath,
+            args: [
+                ...VSCodeIntegrationUtils.pathArgs(this.runConfig),
+                '--new-window',
+                '--inspect',
+                '--skip-release-notes',
+                '--skip-welcome',
+                '--disable-telemetry',
+                '--no-cached-data',
+                '--disable-updates',
+                '--disable-keytar',
+                '--disable-crash-reporter',
+                '--disable-workspace-trust',
+                '--disable-gpu',
+                this.options.workspace
+            ]
+        });
+
+        this.electronApp.on('window', async page => {
+            page.on('pageerror', error => {
+                console.error(error);
+            });
+
+            page.on('console', msg => {
+                // console.log(msg.text());
```

Comment:
I think if we want to keep this line commented, we should at least add a comment above it when we might want to enable this logging (probably a debugging case?). Alternatively, maybe we can enable it based on a generic '--debug' flag but that is not really necessary for this PR.